File: runtime/src/de/fatox/meta/api/dao/MetaProjectData.java
package de.fatox.meta.api.dao;

import com.badlogic.gdx.utils.Array;
import de.fatox.meta.util.StringUtil;

/**
 * Created by Frotty on 02.06.2016.
 */
public class MetaProjectData {
    public static final String PROJECT_FILE_NAME = "metaproject.json";

    public String name = "unknown";
    public Array<MetaTabData> openTabs;

    public MetaProjectData() {
    }

    public MetaProjectData(String name) {
        this.name = name;
        openTabs = new Array<>();
    }

    public boolean isValid() {
        return ! StringUtil.isBlank(name);
    }
}

File: runtime/src/main/kotlin/de/fatox/meta/api/model/MetaProjectData.kt
package de.fatox.meta.api.model

import com.badlogic.gdx.utils.Array

data class MetaProjectData(var name: String = "unknown") {
	var openTabs: Array<MetaTabData> = Array()

	val isValid: Boolean get() = name.isNotBlank()

	companion object {
		const val PROJECT_FILE_NAME: String = "metaproject.json"
	}
}
